### PR TITLE
Login should ignore converter passwords by default

### PR DESCRIPTION
**Pull request by jdarwood007**
By default we shouldn't try to convert non-compatible passwords. Conversions should enable this option automatically to allow proper conversion of passwords.  There is no reason to attempt to fix these passwords for SMF installs that are not conversions and is just a waste of resources to do so.
But we always try to fix old SMF passwords, as there is no reliable way to determine if those options need to be enabled to fix previous bugs, without doing the password checks in the first place.  I suppose we could make this a drop down to also disable SMF ones, but seems worthless.

### CODE REVIEW

File: Sources/LogInOut.php [hunk 1]
```diff
@@ -336,7 +336,8 @@ function Login2()
 			$other_passwords[] = sha1(strtolower($user_settings['member_name']) . un_htmlspecialchars($_POST['passwrd']));
 
 			// BurningBoard3 style of hashing.
-			$other_passwords[] = sha1($user_settings['password_salt'] . sha1($user_settings['password_salt'] . sha1($_POST['passwrd'])));
+			if (!empty($modSettings['enable_password_conversion']))
```
Comment (emanuele45) [inline Sources/LogInOut.php:339]: that one seems not necessary. Am I wrong?

Comment (jdarwood007) [inline Sources/LogInOut.php:339]: I didn't delete it, it was indented.  Burning board is another software so it should follow the same exclusions.  Only SMF passwords from 1.0 (md5), linux-bugged sha1 and pre-htmlchars fix should be tested.

Comment (emanuele45) [inline Sources/LogInOut.php:339]: Yups, sorry, I missed one of the IFs.

**Comment by norv** [comment]
As far as I remember, there's no good way to detect installs that have been converted. Which means converters should be updated at least several months I'd say before this release, to give some time to users of converted boards to login at least once.

**Comment by jdarwood007** [comment]
They get updated on the first login.  So admins should have theirs updated at least to enable the setting if need be. As well emails should be carried over so they can just do a lost password recovery.

**Pull request merged**